05-mvc-with-rails/rails-views-and-controllers/readme.md

```diff
@@ -1,138 +1,325 @@
+## BREAK (10/80)
 
-Since it is horrible practice to configure it the way we did, let's go back and change it back to having security.
+## WE-DO: Artists Create Action (20/100)
+So at this point we have a way to see all artists, view info about a specific artist, as well as see a form to add a new artist.
 
-We're going to remove `config.action_controller.permit_all_parameters = true` from our `config/application.rb`
+But what happens when we try to submit this form?
 
-As well as uncomment `protect_from_forgery with: :exception` in our `app/controllers/application_controller.rb`
+If we were looking in our browser, we would get our old friend `unkown action` error, saying:
+`The action 'create' could not be found for ArtistsController`
 
-What's ultimately going to happen when we update our forms?
+Let's write that action for our controller and get this form to work.
 
-Any time a userviews a form to create, update or destroy a resource, the rails app create a random `authenticity_token` and stores it in a session. When the user than submits the form, rails will look for the authenticity_token compares it to the one stored in the session, and if they match allow the request to continue.
+In `app/controllers/artists_controller.rb`:
+
+```ruby
+# Artists#Create
+def create
+  @artist = Artist.new(params[:artist])
+  if @artist.save
+    redirect_to "/artists/#{@artist.id}"
+  else
+    redirect_to "/artists/new"
+  end
+end
+ ```
+
+With this action, we want to create an instance using the params the user entered into the form, and we also set up a check  to see whether or not the model was stored in our database.
+
+When the request completes, we now need to worry about where to direct the user after adding a new artist.
+
+### Render vs Redirect (5/105)
+Like Sinatra, Rails has many ways to map our app's logic navigating a user's request to a response they care about.
+
+For example, taking a closer look at index action of our Artists Controller:
+
+```ruby
+def index
+  @artists = Artist.all
+end
+```
+
+**Question:** How did we see the artists index view page, when we went to our index route?
+
+If we remember back to Sinatra, you'll notice that we're not explicitly telling the application which view file to render.
+
+That's because rails has implicit rendering. Basically rails is smart enough to know if the artists controller action is called index, then it will look for the index view in the artist folder.
+
+You can explicitly change the implicit render by calling the `render` method in the action of a controller. Something like this:
+
+```ruby
+def index
+  @artists = Artist.all
+  render :index
+end
+```
+
+You could also redirect rather than render. **(ST-WG):** What are the differences between redirecting and rendering?
+
+## (I-DO) Sanitization/Strong Params (10/115)
+On to the task at hand, writing code to handle the `create` action to handle the request from the `new` form.
```

Comment:
Is this a good time to review symbiotic relationship of new/create & edit/update?